src/main/java/com/thealgorithms/strings/AhoCorasick.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Aho-Corasick String Matching Algorithm Implementation
+ *
+ * This code implements the Aho-Corasick algorithm, which is used for efficient
+ * string matching in a given text. It can find multiple patterns simultaneously
+ * and records their positions in the text.
+ *
+ * Author: Prabhat-Kumar-42
+ * GitHub: https://github.com/Prabhat-Kumar-42
+ */
+
+package com.thealgorithms.strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+
+public class AhoCorasick {
+
+    // Trie Node Class
+    public class Node {
+        // Represents a character in the trie
+        private HashMap<Character, Node> child = new HashMap<>(); // Child nodes of the current node
+        private Node suffixLink; // Suffix link to another node in the trie
+        private Node outputLink; // Output link to another node in the trie
+        private int patternInd; // Index of the pattern that ends at this node
+
+        Node() {
+            this.suffixLink = null;
+            this.outputLink = null;
+            this.patternInd = -1;
+        }
+
+        public HashMap<Character, Node> getChild() {
+            return child;
+        }
+
+        public Node getSuffixLink() {
+            return suffixLink;
+        }
+
+        public void setSuffixLink(Node suffixLink) {
+            this.suffixLink = suffixLink;
+        }
+
+        public Node getOutputLink() {
+            return outputLink;
+        }
+
+        public void setOutputLink(Node outputLink) {
+            this.outputLink = outputLink;
+        }
+
+        public int getPatternInd() {
+            return patternInd;
+        }
+
+        public void setPatternInd(int patternInd) {
+            this.patternInd = patternInd;
+        }
+    }
+
+    // Trie Class
+    public class Trie {
+
+        private Node root = null; // Root node of the trie
+        private final String[] patterns; // patterns according to which Trie is constructed
+
+        public Node getTrieRoot() {
+            return root;
+        }

```

Comment:
Is is nowhere used. Is there a reason to have this method? If yes: please write tests for it.

```suggestion
```

Reply:
got it, removed it.